src/Dapr.Client/DaprClient.cs

```diff
@@ -939,6 +940,293 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
             string id,
             CancellationToken cancellationToken = default);
 
+        #region Cryptography
+
+        /// <summary>
+        /// Encrypts an array of bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plainTextBytes">The bytes of the plaintext value to encrypt.</param>
+        /// <param name="algorithm">The name of the algorithm used to wrap the encryption key.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="dataEncryptionCipher">The name of the cipher to use for the encryption operation.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
```

Comment:
Not necessarily for this PR, but I wonder if we should be using the newer .NET 6 (and possibly more appropriate)[`RequiresPreviewFeaturesAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.requirespreviewfeaturesattribute?view=net-6.0) instead of `ObsoleteAttribute` for our alpha APIs.

Reply:
I would handily agree, but I wanted to opt for consistency with how the methods were marked for the distributed lock building block, so I just copied the approach from there.

Reply:
Proposed moving to `RequiresPreviewFeaturesAttribute` in #1219.